app/decorators/user_decorator.rb

```diff
@@ -92,7 +92,7 @@ def qrcode(otp_secret_key)
       digits: TwoFactorAuthenticatable::OTP_LENGTH,
       interval: IdentityConfig.store.totp_code_interval,
     }
-    url = ROTP::TOTP.new(otp_secret_key, options).provisioning_uri(email)
+    url = ROTP::TOTP.new(otp_secret_key, options).provisioning_uri(APP_NAME)
```

Comment:
What is this argument to `provisioning_uri`? I guess I wonder if it's required / redundant if we're also passing `APP_NAME` as `issuer`.

Reply:
It's a name, and is a required argument:
https://github.com/mdp/rotp/blob/2be11551ac4982670d526ceef4a5db985d0f626a/lib/rotp/totp.rb#L51-L58

I'm definitely open to suggestions on what to replace it with

Reply:
Domain is probably the leading idea I have so far.

Reply:
Is this strictly required as part of this work? I guess I think it's at least conventional to include email address or some user identifier, at least based on my own anecdotal observation of the dozens of TOTP configurations I have, even if that's complicated by the fact that a user could have multiple email addresses.

Reply:
After testing a handful (Apple passwords, Authy, Duo, Google Authenticator), both Authy and Google Authenticator use the `name` parameter alongside the issuer. Duo only shows the name. This could be kind of confusing to only show the email, but I'm not sure that we have many great options that can cover all of the apps well.

I've undone the change so we use a confirmed email address for the `name`.